### PR TITLE
refreshes secure markup page

### LINKED ISSUE #288: Revisit secure markup page
Me to revisit the secure markup page, as some looks dated, and some not brilliantly written. Plus there may be some new shiny stuff I can add.
https://github.com/springernature/frontend-playbook/blob/master/security/secure-markup.md

### PR DESCRIPTION
**Pull request by jpw**
Hopefully fixes #288

### CODE REVIEW

File: security/secure-markup.md [hunk 1]
```diff
@@ -34,9 +34,11 @@ Often we serve assets such as JavaScript or CSS files via [CDN](https://www.clou
 
 Files served via CDN's (or your own asset servers) are high-priority targets for attackers, because they enable attackers to inject [payloads](https://en.wikipedia.org/wiki/Payload_(computing)#Security) into many sites with just one breach.
 
-Subresource Integrity protects against corrupted resource files by instructing the browser to generate a base64-encoded cryptographic hash of the file, and comparing that hash to a hash we specify in the markup. If the hashes don't match, the browser doesn't parse the loaded file.
+Subresource Integrity protects against corrupted resource files by instructing the browser to calculate a [hash](https://en.wikipedia.org/wiki/Cryptographic_hash_function) of the downloaded file's content, then compare that hash to one we previously computed for the known-good file. If the hashes don't match, the browser doesn't parse the loaded file.
 
-This may sound complicated but it's actually easy to do!
+We tell the browser the hash of the known-good file by embedding the hash value in the HTML&mdash;using the `integrity` atrribute on the corresponding `script` or `link` element.
```
Comment (josebolos) [inline security/secure-markup.md]: "atrribute"

File: security/secure-markup.md [hunk 2]
```diff
@@ -45,13 +47,11 @@ Example markup (from the [MDN article on Subresource Integrity](https://develope
...
 
-Use of the `crossorigin` attribute depends on [CORS](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS) support of the asset server.
+(Use of the `crossorigin` attribute depends on [CORS](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS) support of the asset server.)
 
 Subresource Integrity can be used in conjunction with a [Content Security Policy](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP) but does not require a CSP to work.
 
-At the time of writing, [browser support for Subresource Integrity](https://caniuse.com/#feat=subresource-integrity) is very good and getting better.
-
-As Subresource Integrity should be relatively easy to add to a front-end build tool chain, it's recommended you use it for any `script` or `link` element.
+As Subresource Integrity should be relatively easy to add to a front-end build tool chain, it's recommended you use it for any `script` or `link` element, especially as [browser support for Subresource Integrity](https://caniuse.com/#feat=subresource-integrity) is good.
```
Comment (hollsk) [inline security/secure-markup.md]:
```suggestion
As Subresource Integrity should be relatively easy to add to a front-end build tool chain, you should use it for any `script` or `link` element, especially as [browser support for Subresource Integrity](https://caniuse.com/#feat=subresource-integrity) is good.
```
When I see something like "it's recommended", I say to myself, "by whom?", and then I replace the passive voice with a direct instruction 😉

File: security/secure-markup.md [hunk 3]
```diff
@@ -34,9 +34,11 @@ Often we serve assets such as JavaScript or CSS files via [CDN](https://www.clou
 
 Files served via CDN's (or your own asset servers) are high-priority targets for attackers, because they enable attackers to inject [payloads](https://en.wikipedia.org/wiki/Payload_(computing)#Security) into many sites with just one breach.
 
-Subresource Integrity protects against corrupted resource files by instructing the browser to generate a base64-encoded cryptographic hash of the file, and comparing that hash to a hash we specify in the markup. If the hashes don't match, the browser doesn't parse the loaded file.
+Subresource Integrity protects against corrupted resource files by instructing the browser to calculate a [hash](https://en.wikipedia.org/wiki/Cryptographic_hash_function) of the downloaded file's content, then compare that hash to one we previously computed for the known-good file. If the hashes don't match, the browser doesn't parse the loaded file.
 
-This may sound complicated but it's actually easy to do!
+We tell the browser the hash of the known-good file by embedding the hash value in the HTML&mdash;using the `integrity` attribute on the corresponding `script` or `link` element.
+
+This may sound complicated but it's not too hard to do!
```
Comment (hollsk) [inline security/secure-markup.md]: This is a lovely rewrite of what was there before 😍 👌

**Comment by hollsk** [comment]
the stuff in the autofill section reminds me that we should probably add a bit about the accessibility of disabling that to the a11y checklist as well (then we can probably link between these two docs to make it clear that there are multiple reasons not to mess with switching off autofill). Will create an issue 👍

**Comment by jpw** [comment]
Thank yous! All done I think.

**Pull request merged**